### DISCUSSION TITLE
troika-three-text: Get width of whitespace

### DISCUSSION THREAD

**Discussion by lockinmarv** [protectwise/troika#257] [Q&A]
Hey! I'm using troika-three-text for creating each word separately for animation purposes.
I've noticed it trims the text if it starts or ends with a space.

Is it possible to get the width of that space somehow? Is there any property I can use within `textRenderInfo` for that?
AFAIK, using just the font size isn't 100% accurate.

I've also tried using the caret utility functions, but I don't really know how that could help.

**Comment by lojjic** [comment]
Hmm... the caret data should include info for leading/trailing spaces. I think you could just ask for the [selection rect](https://protectwise.github.io/troika/troika-three-text/#getselectionrectstextrenderinfo-start-end) for that particular whitespace character, or for the whole string if you want the total word width.

**Reply by lockinmarv** [reply]
Thanks for your answer! So when I use `myText.text = 'Foo '` (with the extra space at the end), it should include that space in the selection rect, but doesn't actually render it? I thought it's completely omitted before it even gets processed.

Is it correct to calculate the total width from the selection rect's left + right like this?
```ts
const word = 'foo '
const text = new Text()
text.text = word
// ...
const rects = getSelectionRects(text.textRenderInfo, 0, word.length)
const width = rects[0].right - rects[0].left
```
It works, but the space isn't 100% correct, it seems. The spacing between "," and "y" is slightly off.
Can this be improved, or is this the best result I can get?

**Whole text:**
<img width="499" alt="CleanShot 2023-05-17 at 20 35 07@2x" src="https://github.com/protectwise/troika/assets/6840287/e4e65811-375b-44c7-9302-4834f9b4818d">

**Separate words:**
<img width="501" alt="CleanShot 2023-05-17 at 20 37 48@2x" src="https://github.com/protectwise/troika/assets/6840287/c7edfbf4-7c74-41e2-9eff-338ab13262a0">

**EDIT:**
My bad, I had some wrong calculations for positioning which halved the whole word length (including the space).
Just getting the half of the word width + adding spacing afterwards works perfectly. 👀😆
